express/blocks/search-marquee/search-marquee.js

```diff
@@ -146,10 +150,24 @@ function initSearchFunction(block) {
         const li = createTag('li', { tabindex: 0 });
         const valRegEx = new RegExp(searchBar.value, 'i');
         li.innerHTML = item.query.replace(valRegEx, `<b>${searchBarVal}</b>`);
-        li.addEventListener('click', () => {
-          if (item.query === searchBar.value) return;
-          searchBar.value = item.query;
-          searchBar.dispatchEvent(new Event('input'));
+        li.addEventListener('click', async () => {
+          if (item.query !== searchBar.value) {
```

Comment:
you probably want to put your duplicate code in a method right?

```
if (item.query !== searchBar.value) {
  searchBar.value = item.query;
  searchBar.dispatchEvent(new Event('input'));
}

await onSearchSubmit();
```

Reply:
Yes!